x-pack/plugin/core/src/main/java/org/elasticsearch/license/LicenseService.java

```diff
@@ -239,36 +235,49 @@ public void registerLicense(final PutLicenseRequest request, final ActionListene
-                        }
+                AckedClusterStateUpdateTask<PutLicenseResponse>(request, listener) {
+                    @Override
+                    protected PutLicenseResponse newResponse(boolean acknowledged) {
+                        return new PutLicenseResponse(acknowledged, LicensesStatus.VALID);
+                    }
 
-                        @Override
-                        public ClusterState execute(ClusterState currentState) throws Exception {
-                            XPackPlugin.checkReadyForXPackCustomMetadata(currentState);
-                            MetaData currentMetadata = currentState.metaData();
-                            LicensesMetaData licensesMetaData = currentMetadata.custom(LicensesMetaData.TYPE);
-                            Version trialVersion = null;
-                            if (licensesMetaData != null) {
-                                trialVersion = licensesMetaData.getMostRecentTrialVersion();
-                            }
-                            MetaData.Builder mdBuilder = MetaData.builder(currentMetadata);
-                            mdBuilder.putCustom(LicensesMetaData.TYPE, new LicensesMetaData(newLicense, trialVersion));
-                            return ClusterState.builder(currentState).metaData(mdBuilder).build();
+                    @Override
+                    public ClusterState execute(ClusterState currentState) throws Exception {
+                        XPackPlugin.checkReadyForXPackCustomMetadata(currentState);
+                        final Version oldestNodeVersion = currentState.nodes().getSmallestNonClientNodeVersion();
+                        if (licenseIsCompatible(newLicense, oldestNodeVersion) == false) {
+                            throw new IllegalStateException("The provided license is not compatible with node version [" +
+                                oldestNodeVersion + "]");
+                        }
+                        MetaData currentMetadata = currentState.metaData();
+                        LicensesMetaData licensesMetaData = currentMetadata.custom(LicensesMetaData.TYPE);
+                        Version trialVersion = null;
+                        if (licensesMetaData != null) {
+                            trialVersion = licensesMetaData.getMostRecentTrialVersion();
                         }
-                    });
+                        MetaData.Builder mdBuilder = MetaData.builder(currentMetadata);
+                        mdBuilder.putCustom(LicensesMetaData.TYPE, new LicensesMetaData(newLicense, trialVersion));
+                        return ClusterState.builder(currentState).metaData(mdBuilder).build();
+                    }
+                });
+        }
+    }
+
+    private static boolean licenseIsCompatible(License license, Version version) {
+        if (License.LicenseType.ENTERPRISE.getTypeName().equalsIgnoreCase(license.type())) {
+            return version.onOrAfter(Version.V_8_0_0);
```

Comment:
IIRC the suggested approach is to use the targeted version here and set bwc_enabled to false in `build.gradle` until the backporting PR is merged